go/jsonclient/client_test.go

```diff
@@ -0,0 +1,304 @@
+package jsonclient
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/google/certificate-transparency/go/testdata"
+)
+
+func TestNewJSONClient(t *testing.T) {
+	var testData = []struct {
+		client *http.Client
+		pubKey string
+		errstr string
+	}{
+		{nil, "bogus", "no PEM block"},
+		{nil, testdata.RsaPublicKeyPEM, ""},
+		{nil, testdata.EcdsaPublicKeyPEM, ""},
+		{nil, testdata.DsaPublicKeyPEM, "Unsupported public key type"},
+		{nil, testdata.RsaPublicKeyPEM + "bogus", "extra data found"},
+	}
+	for i, tc := range testData {
+		client, err := New("http://127.0.0.1", tc.client, tc.pubKey)
+		if tc.errstr == "" {
+			if err != nil {
+				t.Errorf("#%d New() failed: %s", i, err.Error())
```

Comment:
I think this is unnecessary, particularly for error cases where the error itself should be helpful. 

For example, filling in the ellipses in your example gives an obvious redundancy:
```go
  {"key with no PEM block", "bogus", "no PEM block"},
```